test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/CollectionModelBinderTest.cs

```diff
@@ -390,28 +389,26 @@ public void CanCreateInstance_ReturnsExpectedValue(Type modelType, bool expected
 
         private static IModelBinder CreateIntBinder()
         {
-            Mock<IModelBinder> mockIntBinder = new Mock<IModelBinder>();
-            mockIntBinder
-                .Setup(o => o.BindModelAsync(It.IsAny<ModelBindingContext>()))
-                .Returns((ModelBindingContext mbc) =>
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+            return new StubModelBinder(async mbc =>
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
```

Comment:
Could we convert most of these setups to avoid the "`async` but not `await`" mess? E.g. set `mbc.Result` in this lambda.